packages/terra-dropdown-button/src/SplitButton.jsx

```diff
@@ -49,6 +49,17 @@ const propTypes = {
    * The intl object to be injected for translations.
    */
   intl: intlShape.isRequired,
+  /**
+   * @private
+   * Callback to tell the parent it should close the dropdown
+   */
+  requestClose: PropTypes.func,
+  /**
+   *  @private
+   * The associated metaData to be provided in the onSelect callback.
+   */
+  // eslint-disable-next-line react/forbid-prop-types
+  metaData: PropTypes.object,
```

Comment:
Doesn't each item have its associated `metaData` already? Is this specific to just the primary item? I'd not expect `SplitButton` to have metaData per se.

Reply:
yes it is just specific to primary-item because even primary-button is considered as item in SplitButton.

Reply:
Then why do you need a prop here when each item already has metaData prop?

Reply:
yes each item has metadata but we don't have one for primary-button. so we need this prop.

Reply:
In my understanding, primary button is not a separate item - it is built off of one of the items that are passed in as children - and metaData is part of that item. 

Also, 

- if this is the face-up component that the consumers will use to create a SplitButton, then how does this work because it is marked as private? 
- is there an example or a test page that prints out the metaData of the primary button when selected?

Reply:
Updated change-log and tests for metaData usage : ac055df370ea3dc545ea0f06f8c17ef7f24d3620